### PR TITLE
Dịch "adam_vn" - Phần 3

### LINKED ISSUE #1972: Dịch "adam_vn" - Phần 3
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_optimization/adam_vn.md#L122-L221

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by davidnvq**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Điền số issue sau dấu `#` (không có dấu cách): Close #1972 

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn. Bạn đã kiểm tra rằng mỗi đoạn văn đã nằm một dòng?

* [x] Điền tên đầy đủ của bạn vào mục **Những người thực hiện** ở cuối tập tin.

* [ ] Điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Lưu ý cho người phản biện (Reviewer)

1. Chỉ review nếu Pull Request (PR) này đã sẵn sàng cho việc review. Một PR sẵn sàng khi PR đó không có label `status: wip`.
2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

### CODE REVIEW

File: chapter_optimization/adam_vn.md [hunk 1]
```diff
@@ -183,7 +186,8 @@ A more concise implementation is straightforward since `adam` is one of the algo
 Hence we only need to pass configuration parameters for an implementation in Gluon.
 -->
 
-*dịch đoạn phía trên*
+Một cách lập trình súc tích hơn là gọi trực tiếp vì `adam` là một trong những thuật toán được cung cấp sẵn trong thư viện tối ưu `trainer` của Gluon.
```
Comment (cuongvng) [inline chapter_optimization/adam_vn.md:189]:
```suggestion 
Cách lập trình súc tích hơn là gọi trực tiếp `adam` được cung cấp sẵn trong thư viện tối ưu `trainer` của Gluon.
```

File: chapter_optimization/adam_vn.md [hunk 2]
```diff
@@ -195,15 +199,18 @@ d2l.train_gluon_ch11('adam', {'learning_rate': 0.01}, data_iter)
...
 
 <!--
 One of the problems of Adam is that it can fail to converge even in convex settings when the second moment estimate in $\mathbf{s}_t$ blows up.
 As a fix :cite:`Zaheer.Reddi.Sachan.ea.2018` proposed a refined update (and initialization) for $\mathbf{s}_t$.
 To understand what's going on, let us rewrite the Adam update as follows:
 -->
 
-*dịch đoạn phía trên*
+Một trong những vấn đề của Adam là nó có thể không hội tụ ngay cả trong các điều kiện lồi khi ước lượng mô-men bậc hai trong $\mathbf{s}_t$ tăng bùng nổ.
+:cite:`Zaheer.Reddi.Sachan.ea.2018` đề xuất một bản vá với một cập nhật (và khởi tạo) được làm mịn cho $\mathbf{s}_t$.
```
Comment (cuongvng) [inline chapter_optimization/adam_vn.md:212]:
```suggestion
:cite:`Zaheer.Reddi.Sachan.ea.2018` đề xuất làm mịn việc cập nhật (và khởi tạo) $\mathbf{s}_t$.
```

File: chapter_optimization/adam_vn.md [hunk 3]
```diff
@@ -216,7 +223,10 @@ Now the magnitude of the update no longer depends on the amount of deviation.
 This yields the Yogi updates
 -->
 
-*dịch đoạn phía trên*
+Khi phương sai của $\mathbf{g}_t^2$ có giá trị lớn hay các cập nhật là thưa, có thể $\mathbf{s}_t$ sẽ quên các giá trị trước đó quá nhanh.
```
Comment (cuongvng) [inline chapter_optimization/adam_vn.md:226]:
```suggestion
Khi $\mathbf{g}_t^2$ có phương sai lớn hay các cập nhật là thưa, có thể $\mathbf{s}_t$ sẽ quên các giá trị quá khứ quá nhanh.
```

File: chapter_optimization/adam_vn.md [hunk 4]
```diff
@@ -216,7 +223,10 @@ Now the magnitude of the update no longer depends on the amount of deviation.
 This yields the Yogi updates
 -->
 
-*dịch đoạn phía trên*
+Khi $\mathbf{g}_t^2$ có phương sai lớn hay các cập nhật là thưa, có thể $\mathbf{s}_t$ sẽ quên các giá trị quá khứ quá nhanh.
+Một cách giải quyết vấn đề trên đó là thay $\mathbf{g}_t^2 - \mathbf{s}_{t-1}$ bằng $\mathbf{g}_t^2 \odot \mathop{\mathrm{sgn}}(\mathbf{g}_t^2 - \mathbf{s}_{t-1})$.
+Bây giờ, độ lớn của cập nhật không còn phụ thuộc vào giá trị độ lệch.
+Từ đó ta có các cập nhật Yogi sau:
```
Comment (cuongvng) [inline chapter_optimization/adam_vn.md:229]:
```suggestion
Từ đó ta có cập nhật Yogi sau:
```

File: chapter_optimization/adam_vn.md [hunk 5]
```diff
@@ -129,15 +129,17 @@ Last, the explicit learning rate $\eta$ allows us to control the step length to
...
 
 <!--
 Implementing Adam from scratch is not very daunting.
 For convenience we store the timestep counter $t$ in the `hyperparams` dictionary.
 Beyond that all is straightforward.
 -->
 
-*dịch đoạn phía trên*
+Lập trình Adam từ đầu không quá khó khăn.
+Để thuận tiện, chúng ta lưu trữ các biến đếm bước thời gian $t$ trong từ điển `hyperparams`.
```
Comment (minhduc0711) [inline chapter_optimization/adam_vn.md:141]:
```suggestion
Để thuận tiện, chúng ta lưu trữ biến đếm bước thời gian $t$ trong từ điển `hyperparams`.
```

File: chapter_optimization/adam_vn.md [hunk 6]
```diff
@@ -195,15 +199,18 @@ d2l.train_gluon_ch11('adam', {'learning_rate': 0.01}, data_iter)
...
 
 <!--
 One of the problems of Adam is that it can fail to converge even in convex settings when the second moment estimate in $\mathbf{s}_t$ blows up.
 As a fix :cite:`Zaheer.Reddi.Sachan.ea.2018` proposed a refined update (and initialization) for $\mathbf{s}_t$.
 To understand what's going on, let us rewrite the Adam update as follows:
 -->
 
-*dịch đoạn phía trên*
+Một trong những vấn đề của Adam là nó có thể không hội tụ ngay cả trong các điều kiện lồi khi ước lượng mô-men bậc hai trong $\mathbf{s}_t$ tăng bùng nổ.
+:cite:`Zaheer.Reddi.Sachan.ea.2018` đề xuất làm mịn việc cập nhật (và khởi tạo) $\mathbf{s}_t$.
```
Comment (minhduc0711) [inline chapter_optimization/adam_vn.md:212]:
```suggestion
:cite:`Zaheer.Reddi.Sachan.ea.2018` đề xuất phiên bản cải thiện của bước cập nhật (và khởi tạo) $\mathbf{s}_t$ để giải quyết vấn đề này.
```

File: chapter_optimization/adam_vn.md [hunk 7]
```diff
@@ -195,15 +199,18 @@ d2l.train_gluon_ch11('adam', {'learning_rate': 0.01}, data_iter)
...
+Để hiểu rõ hơn, chúng ta hãy viết lại phép cập nhật Adam như sau:
```
Comment (minhduc0711) [inline chapter_optimization/adam_vn.md:213]:
```suggestion
Để hiểu rõ hơn, chúng ta hãy viết lại bước cập nhật Adam như sau:
```

File: chapter_optimization/adam_vn.md [hunk 8]
```diff
@@ -216,7 +223,10 @@ Now the magnitude of the update no longer depends on the amount of deviation.
 This yields the Yogi updates
 -->
 
-*dịch đoạn phía trên*
+Khi $\mathbf{g}_t^2$ có phương sai lớn hay các cập nhật là thưa, có thể $\mathbf{s}_t$ sẽ quên các giá trị quá khứ quá nhanh.
```
Comment (minhduc0711) [inline chapter_optimization/adam_vn.md:226]:
```suggestion
Khi $\mathbf{g}_t^2$ có phương sai lớn hay các cập nhật trở nên thưa, có thể $\mathbf{s}_t$ sẽ quên các giá trị quá khứ quá nhanh.
```

File: chapter_optimization/adam_vn.md [hunk 9]
```diff
@@ -216,7 +223,10 @@ Now the magnitude of the update no longer depends on the amount of deviation.
...
+Một cách giải quyết vấn đề trên đó là thay $\mathbf{g}_t^2 - \mathbf{s}_{t-1}$ bằng $\mathbf{g}_t^2 \odot \mathop{\mathrm{sgn}}(\mathbf{g}_t^2 - \mathbf{s}_{t-1})$.
+Bây giờ, độ lớn của cập nhật không còn phụ thuộc vào giá trị độ lệch.
+Từ đó ta có cập nhật Yogi sau:
```
Comment (minhduc0711) [inline chapter_optimization/adam_vn.md:229]:
```suggestion
Từ đó ta có bước cập nhật Yogi sau:
```

File: chapter_optimization/adam_vn.md (hunk 6 above)
Comment (davidnvq) [reply chapter_optimization/adam_vn.md:212]: làm mịn mình thấy rõ nghĩa hơn nhỉ

Comment (minhduc0711) [reply chapter_optimization/adam_vn.md:212]: mình xem qua công thức yogi nó k gợi lên sự `mịn` lắm, nên mình chỉ hiểu là `cải tiến` thôi

Comment (davidnvq) [reply chapter_optimization/adam_vn.md:238]: từ bài báo
image

Comment (minhduc0711) [reply chapter_optimization/adam_vn.md:238]: đây là assumption hàm mất mát là L-smooth mà, mình thấy không liên quan gì đến thay đổi ở v_t trong yogi
image

Comment (davidnvq) [reply chapter_optimization/adam_vn.md:238]: ô sao không nhỉ, hàm mất mát dc làm mịn, thì gradient cũng sẽ có ảnh hưởng, vt là moment bậc hai cũng sẽ ảnh hưởng theo.  @minhduc0711 nên mở PR bên d2l xem sao ;)

Comment (davidnvq) [reply chapter_optimization/adam_vn.md:238]: ô vậy à, mình cứ hiểu nhầm là khi L đc tác giả assume là L-smooth, thì gradient cũng sẽ ảnh hưởng, và vt cũng sẽ ảnh hưởng theo :-?

Comment (davidnvq) [reply chapter_optimization/adam_vn.md:238]: Anw, nếu mn k chắc là d2l dùng từ refine theo ý gì thì nên dịch chung chung là "cải thiện" ;) mình cũng k quá rành về optimization.
